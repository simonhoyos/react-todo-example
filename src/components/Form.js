import React from 'react';
import { TasksConsumer } from '../Context';

export function Form() {
  return (
    <TasksConsumer>
      {({ title, editingId, handleChange, handleSubmit }) => (
        <form onSubmit={handleSubmit}>
          <label htmlFor="title">Nombre de la tarea</label>
          <input
            type="text"
            id="title"
            name="title"
            onChange={handleChange}
            value={title}
          />
          <button type="submit">{editingId > -1 ? 'Editar tarea' : 'Crear nueva tarea'}</button>
        </form>
      )}
    </TasksConsumer>
  );
}
