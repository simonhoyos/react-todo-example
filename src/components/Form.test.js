import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Form } from './Form';

describe('<Form />', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Form />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should create a new task', () => {
    const handleChange = jest.fn();
    const wrapper = shallow(<Form handleChange={handleChange} />);

    const input = wrapper.find('[data-testid="input-title"]');
    input.simulate('change', {
      currentTarget: {
        name: 'title',
        value: 'This is a test task',
      }
    });

    expect(handleChange).toHaveBeenCalled();
  });
})
