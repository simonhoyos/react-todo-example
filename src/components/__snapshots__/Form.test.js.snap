// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should match snapshot 1`] = `
<form>
  <label
    htmlFor="title"
  >
    Nombre de la tarea
  </label>
  <input
    data-testid="input-title"
    id="title"
    name="title"
    type="text"
  />
  <button
    type="submit"
  >
    Crear nueva tarea
  </button>
</form>
`;
